### ISSUE TITLE
Cannot pass backend flags via emacs variable `agda2-program-args`

### ISSUE THREAD

**Issue by andreasabel** [agda/agda#3235]
I am trying to pass options to GHC when invoking it via the Agda GHC backend from emacs (`C-c C-x C-c`).  The following attempts fail:
- A local `{-# OPTIONS --ghc-flags=... #-}` is not recognized.
- Adding to `agda2-program-args` fails to start agda with error
```
Error: Unrecognized option:
--ghc-flags=...
```

But for some strange reason, the following does not complain:
```
agda --interaction --ghc-flags=-v
```
I cannot reproduce the error `Unrecognized option` on the command line.

**Comment by andreasabel** [comment]
The culprit may be this call to `parseStandardOptions'` in the execution of `Cmd_load`:
https://github.com/agda/agda/blame/a01873e16a4c45280a09156887e968e278ae61fd/src/full/Agda/Interaction/InteractionTop.hs#L1159 .
It does not use handlers for the backend options and complains about unrecognized options.

**Comment by andreasabel** [comment]
Regression: I am not observing a load failure in agda-2.4.2.6 with option `--ghc-flag=-v` in the `agda2-program-args`.

**Comment by UlfNorell** [comment]
A workaround is to add the options as a `FOREIGN` pragma:
```agda
{-# FOREIGN GHC {-# OPTIONS_GHC -v #-} #-}
```

**Comment by nad** [comment]
I think this change broke the Emacs mode for me:
```
Unrecognized options: -M (did you mean --js ?) -1 (did you mean
--js ?) -. (did you mean --js ?) -5 (did you mean --js ?) -G (did
you mean --js ?) -K (did you mean --js ?) -4 (did you mean --js ?)
-0 (did you mean --js ?) -M (did you mean --js ?) -R (did you mean
--js ?) -T (did you mean --js ?) -S (did you mean --js ?)
```
I have included "+RTS", "-M1.5G", "-K40M" and "-RTS" in `agda2-program-args`.

**Comment by andreasabel** [comment]
Haha, I like the `(did you mean --js ?)` suggestion idiot.  As smart as my "smart" phone.

**Comment by vincent-prz** [comment]
I am trying to pass a `--ghc-flag=-dynamic` argument, but getting `Unrecognized option: --ghc-flag=-dynamic` when compiling in Emacs (while in works on the CLI).

Here is my emacs conf (I'm using Spacemacs):
```elisp
     (agda :variables
           agda2-backend "GHC"
           agda2-program-args '("--ghc-flag=-dynamic")
     )
```

Agda version: 2.5.4.2

**Comment by andreasabel** [comment]
@vincent-prz : Well the problem was fixed in 2.6.0 (see milestone).  Thus, you will need to upgrade or use a workaround.
